code/game/machinery/doors/firedoor.dm

```diff
@@ -218,25 +218,31 @@
 			process_results(checked_turf)
 
 /obj/machinery/door/firedoor/proc/register_adjacent_turfs(atom/loc)
+	if(!loc)
+		return
+
 	for(var/dir in GLOB.cardinals)
 		var/turf/checked_turf = get_step(get_turf(loc), dir)
 
 		if(!checked_turf)
 			continue
 		process_results(checked_turf)
 		RegisterSignal(checked_turf, COMSIG_TURF_EXPOSE, .proc/process_results)
-		RegisterSignal(checked_turf, COMSIG_TURF_CALCULATED_ADJACENT_ATMOS, .proc/process_results)
+	RegisterSignal(loc, COMSIG_TURF_CALCULATED_ADJACENT_ATMOS, .proc/process_results)
```

Comment:
if you're gonna do the unregister first do the register first too (I'm not neurotic you're neurotic )

Reply:
I thought it was following the destroy rules of "Initialize, but backwards".

Reply:
Sure.